src/libcommonserver/io/iohelper.cpp

```diff
@@ -276,6 +280,46 @@ bool IoHelper::_checkIfIsHiddenFile(const SyncPath &path, bool &isHidden, IoErro
 
     return true;
 }
+
+bool IoHelper::getRights(const SyncPath &path, bool &read, bool &write, bool &exec, IoError &ioError) noexcept {
+    read = false;
+    write = false;
+    exec = false;
+
+    ItemType itemType;
+    const bool success = getItemType(path, itemType);
+    if (!success) {
+        LOGW_WARN(logger(), L"Failed to get item type: " << Utility::formatIoError(path, itemType.ioError).c_str());
```

Comment:
Not entirely sure, but it is likely that you can remove `.c_str()` at the end of the line, thanks to the latest changes of Hervé on the logger.

Reply:
Perhaps later?
https://infomaniak.atlassian.net/browse/KDESKTOP-1298?atlOrigin=eyJpIjoiYTdhN2FkMzVhZjNjNDc5NGJmOGMzYmVhOWE0MjMwNTEiLCJwIjoiaiJ9